src/app/bungie-api/authenticated-fetch.ts

```diff
@@ -138,6 +139,18 @@ async function handleRefreshTokenError(response: Error | Response): Promise<Toke
     case 400:
     case 401:
     case 403: {
```

Comment:
Won't this still result in a `FatalTokenError` being thrown even if maintenance is in progress, resulting in logouts? My thought is we should only throw a `FatalTokenError` if we haven't already decided it is a server-side Bungie.net error e.g.

``` typescript
try {
  const data = await response.json();
  if (data && data.error === 'server_error') {
    if (data.error_description === 'SystemDisabled') {
      throw new BungieServerError(t('BungieService.Maintenance'));
    } else {
      throw new BungieServerError(
        `Unknown error getting response token: ${data.error}, ${data.error_description}`
      );
    }
  }
} catch (e) {
  if (e.name === 'BungieServerError') {
    throw e;
  }
}
throw new FatalTokenError('Refresh token expired or not valid, status ' + response.status);

...

class BungieServerError extends Error {
  constructor(msg) {
    super(msg);
    this.name = 'BungieServerError';
  }
}
```